zeroconf/_utils/net.py

```diff
@@ -83,6 +83,7 @@ def get_all_addresses_v6() -> List[Tuple[Tuple[str, int, int], int]]:
 
 
 def ip6_to_address_and_index(adapters: List[Any], ip: str) -> Tuple[Tuple[str, int, int], int]:
+    ip = ip[:ip.index('%')]  # Strip scope_id.
```

Comment:
Is it possible it will be passed in without the '%' and we should only strip it if present?

Reply:
Oh certainly, not sure why I thought this was a good idea.